apollo-router/src/services/subgraph_service.rs

```diff
@@ -761,11 +725,410 @@ async fn call_http(
             .to_graphql_error(None),
         )
     }
+    graphql_response
+}
+
+/// Process a single subgraph batch request
+#[instrument(skip(client_factory, context, request))]
+pub(crate) async fn process_batch(
```

Comment:
We did a lot of work to break common functionality out into new subsidiary functions (e.g.: `get_uri_details(), http_response_to_graphql_response()`. At some point I hoped that we'd do enough re-factoring that it would be easy to re-combine the main controlling functions (`process_batch`, `call_single_http`), but it's still not done.

I'll add this to the follow-up issues list.